ext/ffi/callback.rs

```diff
@@ -555,13 +543,31 @@ where
   let v8_value = cb.v8_value;
   let cb = v8::Local::<v8::Function>::try_from(v8_value)?;
 
-  let isolate: *mut v8::Isolate = &mut *scope as &mut v8::Isolate;
-  LOCAL_ISOLATE_POINTER.with(|s| {
-    if s.borrow().is_null() {
-      s.replace(isolate);
+  let isolate_id_result: Result<u32, u32> = LOCAL_ISOLATE_ID.with(|s| {
+    let value = *s.borrow();
+    if value == 0 {
+      let res = ISOLATE_ID_COUNTER.fetch_update(
+        atomic::Ordering::SeqCst,
+        atomic::Ordering::SeqCst,
+        |v| Some(v + 1),
+      )?;
+      s.replace(res);
+      Ok(res)
+    } else {
+      Ok(value)
     }
   });
 
+  if isolate_id_result.is_err() {
+    return Err(generic_error("Failed to update Isolate ID counter"));
+  }
+
+  let isolate_id = isolate_id_result.unwrap();
+
+  if isolate_id == 0 {
```

Comment:
If overflow is a concern, it might be better to use AtomicU64 and skip the overflow checks, as this probably doesn't have to be super-performant.

Reply:
I don't think it's a concern, really. With the `u32::MAX` a program that every second creates 10 Workers that create at least one `UnsafeCallback` would still take more than 13.5 years to overflow this counter.

Still, only for theoretical reasons I figured it's better to add a panic rather than allow a weird theoretical bug to leak through.